src/miniscript/types/extra_props.rs

```diff
@@ -211,6 +231,8 @@ impl Property for ExtData {
             ops_count_nsat: None,
             stack_elem_count_sat: Some(1),
             stack_elem_count_dissat: Some(1),
+            max_sat_size: Some((33, 33)),
+            max_dissat_size: None,
```

Comment:
Should be 33. Same for other hashes

Reply:
I thought hashes were not dissatisfiable ? Sipa implemented this https://github.com/sipa/miniscript/blob/027b422495a8ee3ea3fa7787150389bb65d1b9de/bitcoin/script/miniscript.h#L535-L538 and iirc we did too.

There is, however, a mistake on the `stack_elem_count_dissat` in this case.

Reply:
Made `stack_elem_count_dissat` None, as i really think the mistake is this way around.

Reply:
Agreed

Reply:
I am certain that this is wrong in the sipa implementation. We should have that as 33 for reasons in the previous comment.

Reply:
Okay, that is bug in the sipa implementation as he explicitly considers them being dissatifyable here:

https://github.com/sipa/miniscript/blob/027b422495a8ee3ea3fa7787150389bb65d1b9de/bitcoin/script/miniscript.h#L657 .

Reply:
Heh, good catch. There is definitely no canonical dissatisfaction for an unwrapped hash.

@darosior I think we want the estimation functions to return the worst-case value that is possible under standardness rules. If the fragment is supposed not to be satisfiable (for malleability checks or other reasons) then we should use the `d` flag to indicate that. But if we make it imposible to even estimate these values, it will become harder to work with incomplete fragments or other malleable scripts.

Reply:
> @darosior I think we want the estimation functions to return the worst-case value that is possible under standardness rules.

Does this mean we want this not to be None even if the dissat is malleable ? If so we could have a dissat cost for virtually all fragments here..

> But if we make it imposible to even estimate these values, it will become harder to work with incomplete fragments or other malleable scripts.

Hmm.. That's how [it was before this PR though](https://github.com/rust-bitcoin/rust-miniscript/pull/150/files#diff-0b3a60807b8797a0bce33355b60f14c996d2e7e783359d47f3d97d81eda66737L746-L788) ? Just to be clear (i didn't follow the last IRC discussions very closely): do we want to not ignore malleable dissatisfaction anymore ?

Reply:
> Does this mean we want this not to be None even if the dissat is malleable ? If so we could have a dissat cost for virtually all fragments here..

I don't think there are any non-canonical ones that we are missing here. As you point out, it was a bug in the previous code too.